Update cross validation function

JieXiong9119 opened the pull request:
Fix testing data generation
Add Monte Carlo cross-validation function

JieXiong9119 committed:
Update cross validation function

Fix testing data generation
Add Monte Carlo cross-validation function

JieXiong9119 committed:
Merge branch 'estcp_cross_validation' into cross-validation

JieXiong9119 commented on run/test_run.py:
```diff
@@ -1,12 +1,16 @@
-    )
-
-test_run.whole_process_training_and_testing()
+randomruns = 10
```
Change this number to define the number of runs.

JanghyunJK commented:
not done with the entire review, but is it possible to keep all jupyter notebook files under FDD_RF_Repo\JupyterNotebook folder? I'm mostly referring to this file `run/results/Results.ipynb`. and preferring to keep all jupyter notebooks in one place if these are mostly used for testing (before being reflected in the actual workflow).

JieXiong9119 commented:
> not done with the entire review, but is it possible to keep all jupyter notebook files under FDD_RF_Repo\JupyterNotebook folder? I'm mostly referring to this file `run/results/Results.ipynb`. and preferring to keep all jupyter notebooks in one place if these are mostly used for testing (before being reflected in the actual workflow).

I'm good with that. The only jupyter notebook I used is the same file `run/results/Results.ipynb` - I just added a portion for visualizing the cross-validation results in boxplot.

Commit:
moving rename jupyter notebook

Commit:
typo corrected

Commit:
moving results jupyter notebook

Commit:
removing unnecessary files

let's keep the repo light by
- not including trained model (.sav file)
- any train/test data (.csv file)
- any results file (.csv file)
- also updated gitignore so that these files are ignored when updated.

Commit:
updates on FDD_RF_Modeling module

- typo corrected
- saving location for time stamp index file (from test split) is changed to the results folder

Commit:
Update .gitignore

Commit:
visualization update

Commit:
misc

JanghyunJK commented:
k, I went through all,
- tested on my side
- saw testing data automatically being generated with 20% split
- tried 10 trials with 400 trees and all went well
- also updated figure in slide 102
- suggested some structural changes in this round
- one thing I want to fix later is to change this two same folder of Norfolk into just one folder but anyways
- I think all changes look good, so see if any of my changes are not working for you
- and if all works fine, just merge this branch to `estcp_cross_validation` branch
- and then submit another PR to merge `estcp_cross_validation` to `main`, so that we could discuss this with Liang.

JieXiong9119 committed:
Add function of evaluating detection performance

Calculate testing TPR & FPR (True Positive Rate and False Positive Rate) for future analysis to evaluate detection-only performance
1. We could plot out ROC curve with TPR vs. FPR
2. Then calculate AUC (area under the curve) of the ROC as a metric

JieXiong9119 commented on FDD_RF_Module/FDD_RF_Modeling.py:
```diff
@@ -67,6 +67,19 @@ def CDDR_tot(self, Real_label, Pred_label):
         CDDR_tot = CD / CP
         return CDDR_tot
 
+    def TPR_FPR_tot(self, Real_label, Pred_label):
```
@JanghyunJK I added this part to enable detection performance evaluation, after we have fault-free data included. For now the output for TPR & FPR will be all 1 & 0.

JieXiong9119 reviewed (commented):
> k, I went through all,
> 
> * tested on my side
> * saw testing data automatically being generated with 20% split
> * tried 10 trials with 400 trees and all went well
> * also updated figure in slide 102
> * suggested some structural changes in this round
> * one thing I want to fix later is to change this two same folder of Norfolk into just one folder but anyways
> * I think all changes look good, so see if any of my changes are not working for you
> * and if all works fine, just merge this branch to `estcp_cross_validation` branch
> * and then submit another PR to merge `estcp_cross_validation` to `main`, so that we could discuss this with Liang.

It works good to me. I just commit a small change to enable detection performance evaluation, after we have fault-free data included. For now the output for TPR & FPR will be all 1 & 0.
If you think it's good, we can merge it!

JanghyunJK commented:
awesome! so TPR/FPR are going into the log results now. got it. worked on my computer, so merging!
